### PR TITLE
Feature/best offense

### PR DESCRIPTION
**Pull request by austincarrjones**
## Type Of Change
- [X] New Feature
- [ ] Debugging
- [ ] Refactor/File Clean-up
- [X] Tests
- [ ] Documentation

## Provide A Summary Of Your Change
Built two methods: #best_offense and #worst_offense. Passing the class spec file, but having trouble with stat_tracker (it's not completing the tests for these methods, from what I can tell. Maybe one of you can try after merging). 

## Group Project Check List
- [X] I made test for my code, and made sure they were passing before submitting PR
- [X] I assigned at least 2 group members to review my PR

### CODE REVIEW

File: lib/game_team.rb [hunk 1]
```diff
@@ -89,9 +89,52 @@ def self.worst_coach(season)
       coach_win_percentage.min_by { |coach, percent| percent }.first
     end
 
-    #best_offense
+    def self.total_goals_per_team
```
Comment (zach-bergman) [inline lib/game_team.rb:92]: Nice job building these hashes for the helpers!

File: lib/game_team.rb [hunk 2]
```diff
@@ -89,9 +89,52 @@ def self.worst_coach(season)
...
+      games_per_team
+    end
 
-    #worst offense
+    def self.average_goals_per_game
+      average_gpg = Hash.new(0)
+      all_game_teams.find_all do |game_team|
+        average_gpg[game_team.team_id] = (GameTeam.total_goals_per_team[game_team.team_id].to_f / GameTeam.total_games_per_team[game_team.team_id]).round(2)
+      end
+      average_gpg.delete(0)
```
Comment (zach-bergman) [inline lib/game_team.rb:113]: Is this how you got rid of the 0=>0?

**Review by zach-bergman — APPROVED** [review]
Looks good!

File: spec/game_team_spec.rb [hunk 3]
```diff
@@ -132,5 +132,36 @@
         expect(GameTeam.fewest_tackles("20142015")).to eq("Sporting Kansas City")
       end
     end
+
+    describe "#total_goals_per_team - helper" do
```
Comment (noahdurbin) [inline spec/game_team_spec.rb:136]: I like your testing here.

File: lib/game_team.rb [hunk 4]
```diff
@@ -89,9 +89,52 @@ def self.worst_coach(season)
...
+    def self.total_games_per_team
+      games_per_team = Hash.new(0)
+      all_game_teams.find_all do |game_team|
+        games_per_team[game_team.team_id] += 1
+      end
+      games_per_team
+    end
 
-    #worst offense
+    def self.average_goals_per_game
```
Comment (noahdurbin) [inline lib/game_team.rb:108]: great work!

**Review by noahdurbin — APPROVED** [review]

File: lib/game_team.rb (hunk 2 above)
Comment (austincarrjones) [reply lib/game_team.rb:113]: Yep! Probably a million other ways to do it but I simply removed that specific key/value pair.

**Pull request merged**